### PR TITLE
UPSTREAM: <carry>: OCPBUGS-29520: fix cpu manager cpuset check

### PR DESCRIPTION
**Pull request by MarSik**
This can be squashed to  04070bb40de8dc3570b0ed68ec1debb133e03606 UPSTREAM: : add management support to kubelet

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Workload partitioning makes the separation between reserved and workload cpus more strict. It is therefore expected the reserved cpus are NOT part of the default cpuset and the existing check was overzealous.

First execution of kubelet after reboot never gets here as the cpuset is computed on line 209. However a kubelet restart without reboot skips this code, recovers from state file and runs the check on line 220.

This was uncovered by decoupling the cpu manager state file cleanup from kubelet restart, doing it only once at reboot as part of OCPBUGS-24366

#### Which issue(s) this PR fixes:

Fixes OCPBUGS-29520

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/approve
/remove-label backports/unvalidated-commits
/label backports/validated-commits

**Comment by MarSik** [comment]
While testing I found one more check that was failing for the described case so I had to update the patch.

File: pkg/kubelet/cm/cpumanager/policy_static.go
Comment (ffromani) [inline pkg/kubelet/cm/cpumanager/policy_static.go:220]: is it ok to skip this check entirely when in managed mode? Or do we have an equivalent check?

**Review by ffromani — COMMENTED** [review]
/lgtm

I wish we can have more tests
I won't mind if we get another review

File: pkg/kubelet/cm/cpumanager/policy_static.go
Comment (MarSik) [reply pkg/kubelet/cm/cpumanager/policy_static.go:220]: Yes, the check is not valid for the managed mode. The extra check for not using reserved is lower in the file (totalKnownCPUs == availableCPUs)

**Comment by MarSik** [comment]
the unit test failure k8s.io/kubernetes/pkg/printers: internalversion is definitely not related to this change...

/hold cancel

**Comment by MarSik** [comment]
This currently depends on inclusion of https://github.com/kubernetes/kubernetes/commit/e9b30a0d29b894eb6e2233e063da3600b8fd2825

**Comment by MarSik** [comment]
@soltysh Maciej, can you please tell me how a k8s patch can get here early? The commit I mentioned above (https://github.com/kubernetes/kubernetes/commit/e9b30a0d29b894eb6e2233e063da3600b8fd2825) is needed to unblock the unit tests after the recent leap day (see my k8s follow-up for explanation https://github.com/kubernetes/kubernetes/pull/123910)

**Review by swatisehgal — APPROVED** [review]
/lgtm

**Pull request merged**
